tensorflow/g3doc/hardware/adding_support/index.md

```diff
@@ -0,0 +1,121 @@
+
+New compute hardware is be release frequently and 
+it is of great interest to do machine learning on 
+this hardware. Much of this hardware is designed 
+with machine learning in mind and had features 
+beyond the normal processor. Taking advantage of
+these novel features is important to make the 
+processor be used to its fullest. For example, the
+[Intel Xeon Phi](http://www.intel.com/content/www/us/en/processors/xeon/xeon-phi-detail.html),
+[KnuPath Hermosa](https://www.knupath.com/products/hermosa-processors/),
+[IBM TrueNorth](http://researchweb.watson.ibm.com/articles/brain-chip.shtml), etc.
+Custom support for each of these different processors
+is needed to make TensorFlow capable of using them.
+
+This is a walk through of how to add support for 
+new hardware to TensorFlow. It is assumed that you
+can link C/C++ code that uses your particular hardware
+into TensorFlow. For simplicity, I will link
+code that implement standard BLAS calls.
+
+## TensorFlow Class Structure
+
+The class structure for Devices,
+* [class DeviceBase](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/device_base.h)
+  * [class Device: public DeviceBase](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/device.h)
+    * [class LocalDevice: public Device](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/local_device.h)
+      * [class BaseGPUDevice: public LocalDevice](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/gpu/gpu_device.h)
+      	* [class GPUDevice: public BaseGPUDevice](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/gpu/gpu_device_factory.cc)
+      * [class ThreadPoolDevice: public LocalDevice](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/threadpool_device.h)
+    * [class RemoteDevice: public Device](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/distributed_runtime/remote_device.cc)
+
+
+There is also a structure referred to as a Device Factory,
+* [class DeviceFactory](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/device_factory.h)
+  * [class ThreadPoolDeviceFactory: public Device Factory](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/threadpool_device_factory.cc)
+  * [class BaseGPUDeviceFactory : public DeviceFactory](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/gpu/gpu_device.h)
+    * [class GPUDeviceFactory: public BaseGPUDeviceFactory](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/gpu/gpu_device_factory.cc)
+  * [class GPUCompatibleCPUDeviceFactory: public DeviceFactory](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/gpu/gpu_device_factory.cc)
+
+
+It is clear that TensorFlow is currently built to support 3 major 
```

Comment:
"Distributed" isn't really a device, so I would rewrite as:

"TensorFlow currently supports CPU and GPU devices.  This document will discuss how to add the plumbing to support a different device."